src/devices/Rfid/Mifare/MifareCard.cs

```diff
@@ -0,0 +1,516 @@
+                    AccessSector.WriteAccessBitsWithKeyA | AccessSector.ReadKeyBWithKeyA |
+                    AccessSector.WriteKeyBWithKeyA))
+            {
+                C1 = 0;
+                C2 = 0;
+                C3 = 1;
+            }
+
+            if (accessSector == (AccessSector.WriteKeyAWithKeyB | AccessSector.ReadAccessBitsWithKeyA |
+                    AccessSector.ReadAccessBitsWithKeyB | AccessSector.WriteAccessBitsWithKeyB |
+                    AccessSector.WriteKeyBWithKeyB))
+            {
+                C1 = 0;
+                C2 = 1;
+                C3 = 1;
+            }
+
+            if (accessSector == (AccessSector.ReadAccessBitsWithKeyA |
+                    AccessSector.ReadAccessBitsWithKeyB | AccessSector.WriteAccessBitsWithKeyB))
+            {
+                C1 = 1;
+                C2 = 0;
+                C3 = 1;
+            }
+
+            if (accessSector == (AccessSector.ReadAccessBitsWithKeyA |
+                    AccessSector.ReadAccessBitsWithKeyB))
+            {
+                C1 = 1;
+                C2 = 1;
+                C3 = 1;
+            }
+
+            // Encode the into the 3 bytes
+            byte b6 = (byte)((((~C2) & 0x01) << 7) | (((~C1) & 0x01) << 3));
+            byte b7 = (byte)(((C1) << 7) | (((~C3) & 0x01) << 3));
+            byte b8 = (byte)(((C3) << 7) | ((C2) << 3));
+            return (b6, b7, b8);
+        }
+
+        /// <summary>
+        /// Encode the sector tailer access type for a specific block
+        /// </summary>
+        /// <param name="blockNumber">The block sector to encode</param>
+        /// <param name="accessType">The access type to encode</param>
+        /// <returns>The encoded sector tailer for the specific block</returns>
+        public (byte b6, byte b7, byte b8) EncodeSectorTailer(byte blockNumber, AccessType accessType)
+        {
+            blockNumber = (byte)(blockNumber % 4);
+
+            byte C1 = 0;
+            byte C2 = 0;
+            byte C3 = 0;
+
+            if (accessType == (AccessType.ReadKeyA | AccessType.ReadKeyB | AccessType.WriteKeyA | AccessType.WriteKeyB |
+                    AccessType.IncrementKeyA | AccessType.IncrementKeyB |
+                    AccessType.DecrementTransferRestoreKeyA | AccessType.DecrementTransferRestoreKeyB))
+            {
+                C1 = 0;
+                C2 = 0;
+                C3 = 0;
+            }
+
+            if (accessType == (AccessType.ReadKeyA | AccessType.ReadKeyB))
+            {
+                C1 = 0;
+                C2 = 1;
+                C3 = 0;
+            }
+
+            if (accessType == (AccessType.ReadKeyA | AccessType.ReadKeyB | AccessType.WriteKeyB))
+            {
+                C1 = 1;
+                C2 = 0;
+                C3 = 0;
+            }
+
+            if (accessType == (AccessType.ReadKeyA | AccessType.ReadKeyB | AccessType.WriteKeyB |
```

Comment:
should you create a separate flag for this combination? Or should this perhaps be: `accessType == AccessType.ReadKeyA || accessType == ...`

Reply:
similar question below

Reply:
Unfortunately, only specific conditions adds up. You need the flag to add them but it's not possible to have a simple condition like you describe.